TundraTD/Assets/Scripts/Mobs/MobsBehaviour/MobBehaviour.cs

```diff
@@ -165,5 +163,19 @@ private void HandleAppliedEffects()
                 }
             }
         }
+        
+        public void RespawnMobFromPool(Vector3 position)
+        {
+            // Set mob position
+            var mobTransform = transform;
+            mobTransform.position = position;
+            
+            // Set visual effects
+            gameObject.SetActive(true);
+            MobModel.SetDefaultMaterial();
+            
+            // Set hp, speed & etc 
+            mobModel.SetDefaultValues();
+        }
```

Comment:
I think pooling logic implies that the pool class should handle instantiation logic. For example, the pool class can have two methods such as `Acquire` and `Release` the first of which creates a new object or takes one from the pool and the second places the object into the pool. Also it would be better if the pool were concurrent to not have any problems with multithreading.

Reply:
My implementation of the pooling is to instantiate all mobs from all portals (mobPortals * mobWaves * mobProperty * mobQuantity mobs) and then disabling them. Then we use _mobWave[_currentMobIndex] to get the mob and then respawn it. Respawning means resetting all values (hp, speed, material) and placing the gameobject back on spawner